src/utils/urlValidator.ts

```
@@ -0,0 +1,125 @@
+  /**
+   * Checks if a string is a valid X username (without URL)
+   * @param username The username to validate
+   * @returns True if valid username format
+   */
+  static isValidUsername(username: string): boolean {
+    if (!username || typeof username !== 'string') {
+      return false;
+    }
+    return this.USERNAME_PATTERN.test(username.trim());
+  }
+}
```

Biome (1.9.4)

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)